### ISSUE TITLE
Upgrading from 2.2.0 to 2.3.4 removed accessory

### ISSUE THREAD

**Issue by christophersjchow** [nitaybz/homebridge-sensibo-ac#100]
This seems like the same issue as #97 but raising a new issue as that one got closed.

I upgraded from 2.2.0 to 2.3.4 and my accessory disappeared from HomeKit. I didn't want to try removing my bridge and having to reset everything up as I have a lot of devices, so have tried downgrading.

Downgrading back to 2.2.0 didn't do anything, but downgrading to 2.1.8 succeeded in restoring the accessory. Subsequently upgrading to 2.3.4 again removed it. Here are some logs that might help.

### Logs upgrading 2.1.8 -> 2.3.4

Going straight to 2.3.4 removes the accessory.

```
[1/4/2023, 7:28:35 PM] [SensiboAC] Creating GET request to Sensibo API --->
[1/4/2023, 7:28:35 PM] [SensiboAC] https://home.sensibo.com/api/v2/users/me/pods?fields=id,acState,measurements,remoteCapabilities,room,temperatureUnit,productModel,location,occupancy,smartMode,motionSensors,filtersCleaning,serial,pureBoostConfig,homekitSupported
[1/4/2023, 7:28:36 PM] [SensiboAC] Successful GET response:
[1/4/2023, 7:28:36 PM] [SensiboAC] [{"id":"xxx","temperatureUnit":"C","room":{"uid":"xxx","name":"Nursery","icon":"Livingroom"},"acState":{"timestamp":{"time":"2023-01-04T08:28:36.572673Z","secondsAgo":0},"on":false,"mode":"cool","targetTemperature":23,"temperatureUnit":"C","fanLevel":"quiet","swing":"stopped","horizontalSwing":"stopped"},"location":{"occupancy":"n/a","name":"Default Home","id":"xxx"},"productModel":"skyv2","serial":"xxx","motionSensors":[],"filtersCleaning":{"acOnSecondsSinceLastFiltersClean":1394414,"filtersCleanSecondsThreshold":1080000,"lastFiltersCleanTime":{"time":"2022-07-24T10:30:17Z","secondsAgo":14162299},"shouldCleanFilters":true},"pureBoostConfig":null,"homekitSupported":false,"remoteCapabilities":{"modes":{"cool":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"heat":{"temperatures":{"F":{"isNative":false,"values":[50,52,54,55,57,59,61,63,64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"fan":{"temperatures":{},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"dry":{"temperatures":{},"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"auto":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]}}},"smartMode":null,"measurements":{"time":{"time":"2023-01-04T08:27:06.641308Z","secondsAgo":89},"temperature":26.2,"humidity":70.4,"feelsLike":26.2,"rssi":-62}}]
[1/4/2023, 7:28:36 PM] [SensiboAC] Unregistering Unnecessary Cached Devices:
[1/4/2023, 7:28:36 PM] [SensiboAC] [
  PlatformAccessory {
    _events: [Object: null prototype] {},
    _eventsCount: 0,
    _maxListeners: undefined,
    services: [ [AccessoryInformation], [HeaterCooler], [Switch] ],
    reachable: false,
    context: {
      type: 'AirConditioner',
      deviceId: 'xxx',
      roomName: 'Nursery'
    },
    _associatedHAPAccessory: Accessory {
      _events: [Object: null prototype],
      _eventsCount: 4,
      _maxListeners: undefined,
      displayName: 'Nursery AC',
      UUID: 'xxxx',
      aid: 21,
      _isBridge: false,
      bridged: true,
      bridgedAccessories: [],
      reachable: true,
      category: 1,
      services: [Array],
      shouldPurgeUnusedIDs: true,
      initialized: false,
      controllers: {},
      _setupID: null,
      controllerStorage: [ControllerStorage],
      lastAccessoriesRequest: 0,
      getPrimaryAccessory: [Function (anonymous)],
      configurationChangeDebounceTimeout: undefined,
      lastKnownUsername: undefined,
      bridge: [Bridge],
      [Symbol(kCapture)]: false
    },
    displayName: 'Nursery AC',
    UUID: 'xxxx',
    category: 1,
    _associatedPlugin: 'homebridge-sensibo-ac',
    _associatedPlatform: 'SensiboAC',
    [Symbol(kCapture)]: false
  }
]
```

### Downgrading to 2.1.8

Here are the logs after downgrading to 2.1.8 where the accessory gets added back.

```
[1/4/2023, 7:25:39 PM] [SensiboAC] Creating GET request to Sensibo API --->
[1/4/2023, 7:25:39 PM] [SensiboAC] https://home.sensibo.com/api/v2/users/me/pods?fields=id,acState,measurements,remoteCapabilities,room,temperatureUnit,productModel,location,occupancy,smartMode,motionSensors,filtersCleaning,serial
[1/4/2023, 7:25:41 PM] [SensiboAC] Successful GET response:
[1/4/2023, 7:25:41 PM] [SensiboAC] [{"id":"xxx","temperatureUnit":"C","room":{"uid":"xxx","name":"Nursery","icon":"Livingroom"},"acState":{"timestamp":{"time":"2023-01-04T08:25:40.978173Z","secondsAgo":0},"on":false,"mode":"cool","targetTemperature":23,"temperatureUnit":"C","fanLevel":"quiet","swing":"stopped","horizontalSwing":"stopped"},"location":{"occupancy":"n/a","name":"Default Home","id":"xxx"},"productModel":"skyv2","serial":"xxx","motionSensors":[],"filtersCleaning":{"acOnSecondsSinceLastFiltersClean":1394414,"filtersCleanSecondsThreshold":1080000,"lastFiltersCleanTime":{"time":"2022-07-24T10:30:17Z","secondsAgo":14162123},"shouldCleanFilters":true},"remoteCapabilities":{"modes":{"cool":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"heat":{"temperatures":{"F":{"isNative":false,"values":[50,52,54,55,57,59,61,63,64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"fan":{"temperatures":{},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"dry":{"temperatures":{},"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"auto":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]}}},"smartMode":null,"measurements":{"time":{"time":"2023-01-04T08:25:36.639392Z","secondsAgo":4},"temperature":26.2,"humidity":70.3,"feelsLike":26.2,"rssi":-62}}]
[1/4/2023, 7:25:41 PM] [SensiboAC] Creating New SensiboAC AirConditioner Accessory in the Nursery
[1/4/2023, 7:25:41 PM] [SensiboAC] Adding HeaterCooler Service in the Nursery
[1/4/2023, 7:25:41 PM] [SensiboAC] Adding Sync Button Switch Service in the Nursery
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - AC Active State: false
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Current HeaterCooler State is: OFF
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Target HeaterCooler State is: OFF
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Current Temperature is: 26.2ºC
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Target Cooling Temperature is: 23ºC
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Target Heating Temperature is: 23ºC
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Temperature Display Units is: C
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Current Relative Humidity is: 70.3 %
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - AC Swing is: SWING_DISABLED
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - AC Rotation Speed is: 14%
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Filter Change Indication: CHANGE_FILTER
[1/4/2023, 7:25:41 PM] [SensiboAC] Nursery AC (GET) - Filter Life Level: 0%
```

### Logs upgrading from 2.1.8 -> 2.2.0

Interestingly going from 2.1.8 to 2.2.0 doesn't remove the accessory.

```
[1/4/2023, 7:49:50 PM] [SensiboAC] Creating GET request to Sensibo API --->
[1/4/2023, 7:49:50 PM] [SensiboAC] https://home.sensibo.com/api/v2/users/me/pods?fields=id,acState,measurements,remoteCapabilities,room,temperatureUnit,productModel,location,occupancy,smartMode,motionSensors,filtersCleaning,serial,pureBoostConfig,homekitSupported
[1/4/2023, 7:49:52 PM] [SensiboAC] Successful GET response:
[1/4/2023, 7:49:52 PM] [SensiboAC] [{"id":"xxx","temperatureUnit":"C","room":{"uid":"xxx","name":"Nursery","icon":"Livingroom"},"acState":{"timestamp":{"time":"2023-01-04T08:49:52.165713Z","secondsAgo":0},"on":false,"mode":"cool","targetTemperature":23,"temperatureUnit":"C","fanLevel":"quiet","swing":"stopped","horizontalSwing":"stopped"},"location":{"occupancy":"n/a","name":"Default Home","id":"xxx"},"productModel":"skyv2","serial":"xxx","motionSensors":[],"filtersCleaning":{"acOnSecondsSinceLastFiltersClean":1394414,"filtersCleanSecondsThreshold":1080000,"lastFiltersCleanTime":{"time":"2022-07-24T10:30:17Z","secondsAgo":14163575},"shouldCleanFilters":true},"pureBoostConfig":null,"homekitSupported":false,"remoteCapabilities":{"modes":{"cool":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"heat":{"temperatures":{"F":{"isNative":false,"values":[50,52,54,55,57,59,61,63,64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"fan":{"temperatures":{},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"dry":{"temperatures":{},"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"auto":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]}}},"smartMode":null,"measurements":{"time":{"time":"2023-01-04T08:49:36.663006Z","secondsAgo":15},"temperature":26.3,"humidity":70.6,"feelsLike":26.3,"rssi":-63}}]
[1/4/2023, 7:49:52 PM] [SensiboAC] Adding HeaterCooler Service in the Nursery
[1/4/2023, 7:49:52 PM] [SensiboAC] Adding Sync Button Switch Service in the Nursery
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - AC Active State: false
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Current HeaterCooler State is: OFF
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Target HeaterCooler State is: OFF
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Current Temperature is: 26.3ºC
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Target Cooling Temperature is: 23ºC
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Target Heating Temperature is: 23ºC
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Temperature Display Units is: C
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Current Relative Humidity is: 70.6 %
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - AC Swing is: SWING_DISABLED
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - AC Rotation Speed is: 14%
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Filter Change Indication: CHANGE_FILTER
[1/4/2023, 7:49:52 PM] [SensiboAC] Nursery AC (GET) - Filter Life Level: 0%
```

Subsequently if I go from 2.2.0 with the accessory present to 2.3.4 it gets removed again:

```
[1/4/2023, 7:55:55 PM] [SensiboAC] Successful GET response:
[1/4/2023, 7:55:55 PM] [SensiboAC] [{"id":"xxxx","temperatureUnit":"C","room":{"uid":"xxx","name":"Nursery","icon":"Livingroom"},"acState":{"timestamp":{"time":"2023-01-04T08:55:55.625932Z","secondsAgo":0},"on":false,"mode":"cool","targetTemperature":23,"temperatureUnit":"C","fanLevel":"quiet","swing":"stopped","horizontalSwing":"stopped"},"location":{"occupancy":"n/a","name":"Default Home","id":"xxx"},"productModel":"skyv2","serial":"xxx","motionSensors":[],"filtersCleaning":{"acOnSecondsSinceLastFiltersClean":1394414,"filtersCleanSecondsThreshold":1080000,"lastFiltersCleanTime":{"time":"2022-07-24T10:30:17Z","secondsAgo":14163938},"shouldCleanFilters":true},"pureBoostConfig":null,"homekitSupported":false,"remoteCapabilities":{"modes":{"cool":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"heat":{"temperatures":{"F":{"isNative":false,"values":[50,52,54,55,57,59,61,63,64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"fan":{"temperatures":{},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"dry":{"temperatures":{},"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"auto":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]}}},"smartMode":null,"measurements":{"time":{"time":"2023-01-04T08:55:36.667498Z","secondsAgo":18},"temperature":26.3,"humidity":70.6,"feelsLike":26.3,"rssi":-63}}]
[1/4/2023, 7:55:55 PM] [SensiboAC] Unregistering Unnecessary Cached Devices:
[1/4/2023, 7:55:55 PM] [SensiboAC] [
  PlatformAccessory {
    _events: [Object: null prototype] {},
    _eventsCount: 0,
    _maxListeners: undefined,
    services: [ [AccessoryInformation], [HeaterCooler], [Switch] ],
    reachable: false,
    context: {
      type: 'AirConditioner',
      deviceId: 'xxx',
      roomName: 'Nursery'
    },
    _associatedHAPAccessory: Accessory {
      _events: [Object: null prototype],
      _eventsCount: 4,
      _maxListeners: undefined,
      displayName: 'Nursery AC',
      UUID: 'xxx',
      aid: 22,
      _isBridge: false,
      bridged: true,
      bridgedAccessories: [],
      reachable: true,
      category: 1,
      services: [Array],
      shouldPurgeUnusedIDs: true,
      initialized: false,
      controllers: {},
      _setupID: null,
      controllerStorage: [ControllerStorage],
      lastAccessoriesRequest: 0,
      getPrimaryAccessory: [Function (anonymous)],
      configurationChangeDebounceTimeout: undefined,
      lastKnownUsername: undefined,
      bridge: [Bridge],
      [Symbol(kCapture)]: false
    },
    displayName: 'Nursery AC',
    UUID: 'xxxx',
    category: 1,
    _associatedPlugin: 'homebridge-sensibo-ac',
    _associatedPlatform: 'SensiboAC',
    [Symbol(kCapture)]: false
  }
]
```

**Comment by benwebbbenwebb** [comment]
Hi @christophersjchow 

Thanks for all the detailed info and sorry for the trouble!

I'm having a look and, given the logs provided, I'm guessing it's something in this check that is causing the issue...

```javascript
case 'AirConditioner':
	deviceExists = platform.devices.find(device => {
		return device.id === accessory.context.deviceId && device.remoteCapabilities && (['sky','skyv2','skyplus','air','airq'].includes(device.productModel) || device.productModel.includes('air') || device.productModel.includes('sky'))
	})
	if (!deviceExists)
		accessoriesToRemove.push(accessory)
	break
```

You've blanked them out in your logs/payloads, which is fine, but are id [device.id] (from the API response) and deviceId from logs (when removing cached accessory) the same?

Also, after doing the upgrade, and seeing the device be removed, is there any chance you could restart the Homebridge service? I would expect, given it's in the Sensibo API response, that if it was no longer in your accessory list it should then get re-added (not a great solution, but hopefully better than having to completely reset Homebridge).

**Comment by christophersjchow** [comment]
Sorry for the delay replying. I've checked the device IDs and they are the same between the API response and the value in `accessory.context.deviceId`. To double check your logic I wrote a node test with the payload response and the code above, which evaluated correctly, finding the accessory. Is there someone where else earlier that pushes to `accessoriesToRemove`?

> Also, after doing the upgrade, and seeing the device be removed, is there any chance you could restart the Homebridge service? I would expect, given it's in the Sensibo API response, that if it was no longer in your accessory list it should then get re-added (not a great solution, but hopefully better than having to completely reset Homebridge).

So the accessory doesn't actually get removed after updating the version. It's only after I restart the homebridge process and it reloads the plugins that the logs show the removal of the accessory. Interestingly the logs continue to show the API requests happening even after the accessory has been removed - it however is never re-registered.

```
[1/9/2023, 10:23:35 PM] [SensiboAC] Successful GET response:
[1/9/2023, 10:23:35 PM] [SensiboAC] [{"id":"xxxx","temperatureUnit":"C","room":{"uid":"xxx","name":"Nursery","icon":"Livingroom"},"acState":{"timestamp":{"time":"2023-01-09T11:23:35.091081Z","secondsAgo":0},"on":true,"mode":"cool","targetTemperature":23,"temperatureUnit":"C","fanLevel":"quiet","swing":"stopped","horizontalSwing":"stopped"},"location":{"occupancy":"n/a","name":"Default Home","id":"bTw7h9DAR5"},"productModel":"skyv2","serial":"xxxx","motionSensors":[],"filtersCleaning":{"acOnSecondsSinceLastFiltersClean":1524459,"filtersCleanSecondsThreshold":1080000,"lastFiltersCleanTime":{"time":"2022-07-24T10:30:17Z","secondsAgo":14604798},"shouldCleanFilters":true},"pureBoostConfig":null,"homekitSupported":false,"remoteCapabilities":{"modes":{"cool":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"heat":{"temperatures":{"F":{"isNative":false,"values":[50,52,54,55,57,59,61,63,64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"fan":{"temperatures":{},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"dry":{"temperatures":{},"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"auto":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]}}},"smartMode":null,"measurements":{"time":{"time":"2023-01-09T11:22:43.098910Z","secondsAgo":52},"temperature":25.9,"humidity":67.7,"feelsLike":25.9,"rssi":-63}}]
[1/9/2023, 10:23:35 PM] [SensiboAC] Unregistering Unnecessary Cached Devices:
[1/9/2023, 10:23:35 PM] [SensiboAC] [
  PlatformAccessory {
    _events: [Object: null prototype] {},
    _eventsCount: 0,
    _maxListeners: undefined,
    services: [ [AccessoryInformation], [HeaterCooler], [Switch] ],
    reachable: false,
    context: {
      type: 'AirConditioner',
      deviceId: 'xxx',
      roomName: 'Nursery'
    },
    _associatedHAPAccessory: Accessory {
      _events: [Object: null prototype],
      _eventsCount: 4,
      _maxListeners: undefined,
      displayName: 'Nursery AC',
      UUID: '71395c23-8beb-45f1-858f-816798a5f73d',
      aid: 23,
      _isBridge: false,
      bridged: true,
      bridgedAccessories: [],
      reachable: true,
      category: 1,
      services: [Array],
      shouldPurgeUnusedIDs: true,
      initialized: false,
      controllers: {},
      _setupID: null,
      controllerStorage: [ControllerStorage],
      lastAccessoriesRequest: 0,
      getPrimaryAccessory: [Function (anonymous)],
      configurationChangeDebounceTimeout: undefined,
      lastKnownUsername: undefined,
      bridge: [Bridge],
      [Symbol(kCapture)]: false
    },
    displayName: 'Nursery AC',
    UUID: '71395c23-8beb-45f1-858f-816798a5f73d',
    category: 1,
    _associatedPlugin: 'homebridge-sensibo-ac',
    _associatedPlatform: 'SensiboAC',
    [Symbol(kCapture)]: false
  }
]
[1/9/2023, 10:25:05 PM] [SensiboAC] Creating GET request to Sensibo API --->
[1/9/2023, 10:25:05 PM] [SensiboAC] https://home.sensibo.com/api/v2/users/me/pods?fields=id,acState,measurements,remoteCapabilities,room,temperatureUnit,productModel,location,occupancy,smartMode,motionSensors,filtersCleaning,serial,pureBoostConfig,homekitSupported
[1/9/2023, 10:25:06 PM] [SensiboAC] Successful GET response:
[1/9/2023, 10:25:06 PM] [SensiboAC] [{"id":"xxxx","temperatureUnit":"C","room":{"uid":"xxx","name":"Nursery","icon":"Livingroom"},"acState":{"timestamp":{"time":"2023-01-09T11:25:06.458639Z","secondsAgo":0},"on":true,"mode":"cool","targetTemperature":23,"temperatureUnit":"C","fanLevel":"quiet","swing":"stopped","horizontalSwing":"stopped"},"location":{"occupancy":"n/a","name":"Default Home","id":"bTw7h9DAR5"},"productModel":"skyv2","serial":"xxxx","motionSensors":[],"filtersCleaning":{"acOnSecondsSinceLastFiltersClean":1524459,"filtersCleanSecondsThreshold":1080000,"lastFiltersCleanTime":{"time":"2022-07-24T10:30:17Z","secondsAgo":14604889},"shouldCleanFilters":true},"pureBoostConfig":null,"homekitSupported":false,"remoteCapabilities":{"modes":{"cool":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"heat":{"temperatures":{"F":{"isNative":false,"values":[50,52,54,55,57,59,61,63,64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"fan":{"temperatures":{},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"dry":{"temperatures":{},"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"auto":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]}}},"smartMode":null,"measurements":{"time":{"time":"2023-01-09T11:24:13.099360Z","secondsAgo":53},"temperature":25.9,"humidity":67.5,"feelsLike":25.9,"rssi":-65}}]
[1/9/2023, 10:26:36 PM] [SensiboAC] Creating GET request to Sensibo API --->
[1/9/2023, 10:26:36 PM] [SensiboAC] https://home.sensibo.com/api/v2/users/me/pods?fields=id,acState,measurements,remoteCapabilities,room,temperatureUnit,productModel,location,occupancy,smartMode,motionSensors,filtersCleaning,serial,pureBoostConfig,homekitSupported
[1/9/2023, 10:26:37 PM] [SensiboAC] Successful GET response:
[1/9/2023, 10:26:37 PM] [SensiboAC] [{"id":"xxxx","temperatureUnit":"C","room":{"uid":"xxx","name":"Nursery","icon":"Livingroom"},"acState":{"timestamp":{"time":"2023-01-09T11:26:37.502637Z","secondsAgo":0},"on":true,"mode":"cool","targetTemperature":23,"temperatureUnit":"C","fanLevel":"quiet","swing":"stopped","horizontalSwing":"stopped"},"location":{"occupancy":"n/a","name":"Default Home","id":"bTw7h9DAR5"},"productModel":"skyv2","serial":"xxxx","motionSensors":[],"filtersCleaning":{"acOnSecondsSinceLastFiltersClean":1524459,"filtersCleanSecondsThreshold":1080000,"lastFiltersCleanTime":{"time":"2022-07-24T10:30:17Z","secondsAgo":14604980},"shouldCleanFilters":true},"pureBoostConfig":null,"homekitSupported":false,"remoteCapabilities":{"modes":{"cool":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"heat":{"temperatures":{"F":{"isNative":false,"values":[50,52,54,55,57,59,61,63,64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"fan":{"temperatures":{},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"dry":{"temperatures":{},"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]},"auto":{"temperatures":{"F":{"isNative":false,"values":[64,66,68,70,72,73,75,77,79,81,82,84,86]},"C":{"isNative":true,"values":[18,19,20,21,22,23,24,25,26,27,28,29,30]}},"fanLevels":["quiet","low","medium_low","medium","medium_high","high","auto","strong"],"swing":["stopped","rangeFull"],"horizontalSwing":["stopped","rangeFull"]}}},"smartMode":null,"measurements":{"time":{"time":"2023-01-09T11:25:43.090865Z","secondsAgo":54},"temperature":25.9,"humidity":67.7,"feelsLike":25.9,"rssi":-64}}]
```

**Comment by benwebbbenwebb** [comment]
Hi @christophersjchow 

Thanks for the additional info.

The removal of previously cached accessories was written before I joined the project so unfortunately I'm not that familiar with it.

The only other lines that add an accessory to the `accessoriesToRemove[]` array are wrapped in qualifiers such as:
```javascript
if (!accessory.context.type) {
	platform.log.easyDebug(`Old cached accessory to be removed, name: ${accessory.displayName}`)
	accessoriesToRemove.push(accessory)
}
```
or other switch cases, e.g.:
```javascript
case 'AirPurifier':
...
```
and with your unit being of `accessory.context.type = 'AirConditioner'` it shouldn't match those...

I also took a look at the diffs between 2.1.8, 2.2.0 and 2.3.4 (the specific file is sensibo/syncHomeKitCache.js). For example https://github.com/nitaybz/homebridge-sensibo-ac/compare/v2.1.8...v2.3.4 and couldn't find much that would account for this behaviour. I know that's not much help to you!

If you're willing there is an alpha version, 2.4.0-alpha.2, to which I've included some more debug logging when an accessory gets removed, it might give more context on what's going on.

Failing that, my next best suggestion, rather than having to reset all of Homebridge, is to remove just the offending accessory from the cache manually. If you are using Homebridge UI (homebridge-config-ui-x) on top of your Homebridge install you can navigate to http://\<homebridgeinstance\>/settings in the browser and then click the option next to 'Remove Single Cached Accessory', from the list remove the AC unit. Once that's done then hopefully the Accessory will be re-added correctly from the API response.

**Comment by christophersjchow** [comment]
Removing the cached accessory worked! After a restart on accessory was added back in and I just had to reallocate the room.

**Comment by benwebbbenwebb** [comment]
Thanks for the update @christophersjchow, glad to hear it worked!

I'll include some similar info in the Readme in the next release to help others if they see a similar issue.
